networkd/network_server.ml

```diff
@@ -999,24 +1001,43 @@ module Bridge = struct
 end
 
 module Sriov = struct
-	open Xcp_pci
 
 	let enable _ dbg ~name =
 		Debug.with_thread_associated dbg (fun () ->
 			debug "Enable NET-SRIOV by name: %s" name;
-			Ok Modprobe_successful
+			match Network_utils.Sriov.enable_internal name with
+			| Ok t -> Ok t
+			| Result.Error (_, msg) -> Error (Msg msg)
 		) ()
 
 	let disable _ dbg ~name =
 		Debug.with_thread_associated dbg (fun () ->	
 			debug "Disable NET-SRIOV by name: %s" name;
-			Ok Modprobe_successful
+			match Network_utils.Sriov.disable_internal name with
+			| Ok t -> Ok Disable_successful
+			| Result.Error (_, msg) -> Error (Msg msg)
 		) ()
 
+	let make_vf_conf_internal pcibuspath (vf_info : Sriov.sriov_pci_t) =
+		let exe_except_none f = function
+			| None -> Result.Ok ()
+			| Some a -> f a
+		in
+		let vlan = Opt.map Int64.to_int vf_info.vlan
+		and rate = Opt.map Int64.to_int vf_info.rate in
+		Sysfs.parent_device_of_vf pcibuspath >>= fun dev ->
+		Sysfs.device_index_of_vf dev pcibuspath >>= fun index ->
+		exe_except_none (Ip.set_vf_mac dev index) vf_info.mac >>= fun () ->
+		exe_except_none (Ip.set_vf_vlan dev index) vlan >>= fun () ->
+		exe_except_none (Ip.set_vf_rate dev index) rate >>= fun () ->
+		Ok ()
+
 	let make_vf_config _ dbg ~pci_address ~vf_info =
 		Debug.with_thread_associated dbg (fun () ->	
-			let pcibuspath = string_of_address pci_address in
+			let pcibuspath = Xcp_pci.string_of_address pci_address in
 			debug "Config VF with pci address: %s" pcibuspath;
+			make_vf_conf_internal pcibuspath vf_info;
+			()
```

Comment:
`make_vf_conf_internal` returns a `result`, but this function throws the return value away and aways returns `()`. I would expect something like this:
```
match make_vf_conf_internal pcibuspath vf_info with
| Ok () -> ()
| Error e -> (* raise some exception *)
```
Or even change the return type of `make_vf_conf` in the the IDL to be a `result` type, and possibly translate the result types in a match as above.

Reply:
Ok. I will change the return type of make_vf_conf in the IDL to be a result to keep the consistency with the enable/disable.